### ISSUE TITLE
Missing "if" break the chart

### ISSUE THREAD

**Issue by pierluigilenoci** [kubecost/cost-analyzer-helm-chart#790]
Around this line, it is necessary to add an if block otherwise an error is generated if there are no accessory labels. 
https://github.com/kubecost/cost-analyzer-helm-chart/blob/3a4ff337a7d164c4f710f45e77f9a192b504e715/cost-analyzer/templates/cost-analyzer-network-costs-template.yaml#L19

Something like
```yaml
    metadata:
      labels:
        app: {{ template "cost-analyzer.networkCostsName" . }}
{{- if .Values.networkCosts.additionalLabels }}
        {{ toYaml .Values.networkCosts.additionalLabels | indent 8 }}
{{- end }}
    spec:
```

Helm error messages:
```
installation failed: DaemonSet.apps \"cost-analyzer-network-costs\" is invalid: [spec.template.metadata.labels: Invalid value: map[string]string{\"app\":\"cost-analyzer-network-costs {}\"}: `selector` does not match template `labels`, spec.template.labels: Invalid value: \"cost-analyzer-network-costs {}\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]
```

**Comment by andreb89** [comment]
Yes, we got the same error. But even setting the labels does not work. 

Instead of `{{ toYaml .Values.networkCosts.additionalLabels | indent 8 }}`
it needs to be: `{{ toYaml .Values.networkCosts.additionalLabels | nindent 8 }}`

Otherwise you are getting:
`
Error: YAML parse error on cost-analyzer/templates/cost-analyzer-network-costs-template.yaml: error converting YAML to JSON: yaml: line 21: mapping values are not allowed in this context
helm.go:81: [debug] error converting YAML to JSON: yaml: line 21: mapping values are not allowed in this context`

**Comment by pierluigilenoci** [comment]
@andreb89 I confirm! @AjayTripathy could you please fix it ASAP?

**Comment by dbpolito** [comment]
Just FYI i tried to add a temporary label to try to workaround this issue and it didn't work too:

```yaml
networkCosts:
  enabled: true
  additionalLabels:
    somethingtemporary: toberemoved
```

I think the indentation is wrong there too... maybe change to `nindent` or remove indentation as other parts of the chart:

https://github.com/kubecost/cost-analyzer-helm-chart/blob/6232d779ec0a686b753073f783c15f75e305a4cc/cost-analyzer/charts/prometheus/charts/kube-state-metrics/templates/servicemonitor.yaml#L13

**Comment by AjayTripathy** [comment]
Very sorry about that. #791 has the fix.

**Comment by andreb89** [comment]
@AjayTripathy Thanks for the fix. If possible, please release a hotfix version. Since the current version has some fixes we need to further test the Azure out of cluster costs.

**Comment by AjayTripathy** [comment]
@andreb89 yes, hotfix scheduled for this afternoon PST.

**Comment by AjayTripathy** [comment]
These issues should be resolved in 1.75.1. Thanks for the reports!

**Comment by dwbrown2** [comment]
Should be resolved, but please reopen if there's more here!
